### DISCUSSION TITLE
TPI3 Conexión con API

### DISCUSSION THREAD

**Discussion by ivanscardaoni1** [dds-jv/foro#290] [Consultas de objetos] [answered]
Buenas! Tengo una duda respecto a cuando utilizar el patrón Adapter. En el caso del TPI3, me resulta muy sencilla la interfaz que utilizan las APIS para mi dominio, serian fácilmente utilizables con una instancia de la clase GugleMapas por ejemplo. Pero a su vez no se si tengo que tener en cuenta el hecho de que en el futuro tranquilamente el GugleMapas podría cambiar a un Waze, y mi App estaría recontra acoplada al GugleMapas. 

La pregunta seria ¿Debo tener en cuenta en los parciales y en este TP el hecho del acoplamiento a componentes externos?¿O asumo que probablemente se mantenga ese (a menos que me digan otra cosa) y si la interfaz es utilizable en mi dominio ni hago un adapter?

Desde ya muchas gracias!

**Comment by gastonprieto** [comment, marked as answer]
Buenas @ivanscardaoni1 

En general tratamos de desacoplarnos de los componentes externos, para ganar en testeabilidad y flexibilidad si es que hay que hacer un cambio, además de darnos un punto donde podamos jugar con la adaptación. En este caso justo el código te dice que es una interfaz por lo tanto podrías usar facilmente mockito para generar el impostor, pero en el enunciado dice que es un componente, ahí no sabes ni la interfaz del componente, ni siquiera si es una clase o interfaz. 

Creo que para el código que esta en el repo podes decidir vos como lo queres hacer en base a esas ventajas/desventajas de meter un objeto extra.
En cambio para el enunciado (sin el código ese que recibiste del repo), hubiera sido necesario modelar tu propio componente, porque te falta mas información, entonces podes acoplar tu solución a tu componente y luego cuando tengas (nunca) el componente meterias las adaptaciones o no que fueran necesarias.

**Reply by ivanscardaoni1** [reply]
Perfecto Gaston! muchas gracias.
